### PR TITLE
Handle dynamic rules object

### LINKED ISSUE #23: LF form should accept computed rules object
Currently read only once, on init: see https://github.com/jbandura/ember-legit-forms/blob/master/addon/components/lf-form.js#L34 .

### PR DESCRIPTION
**Pull request by bardzusny**
* LFForm component: `formValidator` property as computed property, instead of static instance with parameters changed in `dataChanged` observer.
* LFForm component: Adding `rulesChanged` observer, forcing form re-validation when new validation rules are supplied.
* LFInput mixin: altering re-validation handlers to accept option specifying whether all form inputs should be forced to show their validation state (even unmodified ones).
* LFForm integration spec:  _de facto_ test case for all of the above.

TL;DR: `rules` object can be computed now - making dynamic validation much easier and more powerful, without writing custom validators (assuming form values are held as controller/component properties and can be accessed by computed property).

Fixes #23 .

### CODE REVIEW

**Comment by jbandura** [comment]
LGTM, thanks @bardzusny! :)
